ansible_collections/arista/avd/plugins/action/eos_validate_state_runner.py

```diff
@@ -104,6 +108,34 @@ def run(self, tmp=None, task_vars=None):
         return result
 
 
+def get_custom_anta_catalogs(hostvars: Mapping, hostname: str, custom_anta_catalogs_dir: Path) -> list[Path] | None:
+    """Retrieve the custom ANTA catalogs for the current inventory device.
+
+    Custom catalogs can be provided for each device or for each Ansible inventory group of devices.
+
+    They must be named after the device hostname or the group name and have a .yml or .yaml extension.
+
+    Args:
+    ----
+      hostvars (Mapping): The Ansible Hostvars Manager object where we can retrieve hostvars for each host on-demand.
+      hostname (str): Current inventory device that is running the plugin.
+      custom_anta_catalogs_dir (Path): The directory where the custom ANTA catalogs are stored.
+
+    Returns:
+    -------
+      list[Path] | None: The list of custom ANTA catalog files for the current inventory device or None if no custom catalogs are found.
+    """
+    # Get the groups for the current inventory device
+    inventory_manager = hostvars._inventory
+    host = inventory_manager.get_host(hostname)
+    host_groups = {group.get_name() for group in host.get_groups()}
+
+    # Search for custom ANTA catalogs
+    search_prefix = [*list(host_groups), hostname]
+    custom_anta_catalogs = [file for prefix in search_prefix for ext in ("yml", "yaml") for file in custom_anta_catalogs_dir.glob(f"{prefix}.{ext}")]
```

Comment:
Using a generator expression instead of a list comprehension when searching for custom ANTA catalogs. This can potentially save memory if there are a lot of files to search through

```
custom_anta_catalogs = list(file for prefix in search_prefix for ext in ("yml", "yaml") for file in custom_anta_catalogs_dir.glob(f"{prefix}.{ext}"))
```

Reply:
if you call list this is the same

Reply:
Unless I'm missing something I think a `list[Path]` is fine.